content/en/docs/devops-user-guide/how-to-use/create-a-pipeline-using-jenkinsfile.md

```diff
@@ -0,0 +1,289 @@
+
+## Prerequisites
+
+- You need to have a DokcerHub account and a GitHub account.
+- You need to create a workspace, a DevOps project, and a **project-regular** user account, and this account needs to be invited into a DevOps project.
+- Set CI dedicated node for building pipeline, please refer to [Set CI Node for Dependency Cache](../../how-to-use/set-ci-node/).
+- You need to install and configure sonarqube, please refer to [How to integrate SonarQube in Pipeline
+](../../../how-to-integrate/sonarqube/) . Or you can skip this part, There is no **Sonarqube Analysis** below.
+
+## Pipeline Overview
+
+There are eight stages as shown below in the pipeline that is going to demonstrate.
+
+![Pipeline Overview](https://pek3b.qingstor.com/kubesphere-docs/png/20190512155453.png#align=left&display=inline&height=1302&originHeight=1302&originWidth=2180&search=&status=done&width=2180)
+
+> Note：
+
+> - **Stage 1. Checkout SCM**: Checkout source code from GitHub repository.
+> - **Stage 2. Unit test**: It will continue to execute next stage after unit test passed.
+> - **Stage 3. SonarQube analysis**：Process sonarQube code quality analysis.
+> - **Stage 4.** **Build & push snapshot image**: Build the image based on selected branches in the behavioral strategy. Push the tag of `SNAPSHOT-$BRANCH_NAME-$BUILD_NUMBER` to DockerHub, among which, the `$BUILD_NUMBER` is the operation serial number in the pipeline's activity list.
+> - **Stage 5. Push the latest image**: Tag the sonarqube branch as latest and push it to DockerHub.
+> - **Stage 6. Deploy to dev**: Deploy sonarqube branch to Dev environment. verification is needed for this stage.
+> - **Stage 7. Push with tag**: Generate tag and released to GitHub. Then push the tag to DockerHub.
+> - **Stage 8. Deploy to production**: Deploy the released tag to the Production environment.
+
+## Hands-on Lab
+
+### Step 1: Create Credentials
+
+> Note: If there are special characters in your account or password, please encode it using https://www.urlencoder.org/, then paste the encoded result into credentials below.
+
+1.1. Log in KubeSphere with the account `project-regular`, enter into the created DevOps project and create the following three credentials under **Project Management → Credentials**:
+
+|Credential ID| Type | Where to use |
+| --- | --- | --- |
+| dockerhub-id | Account Credentials | DockerHub |
+| github-id | Account Credentials | GitHub |
+| demo-kubeconfig | kubeconfig | Kubernetes |
+
+1.2. We need to create an additional credential `sonar-token` for SonarQube token, which is used in stage 3 (SonarQube analysis) mentioned above. Refer to [Access SonarQube Console and Create Token](../../how-to-integrate/sonarqube/) to copy the token and paste here. Then press **OK** button.
```

Comment:
The expression "copy the token and paste here" is not clear. I noticed this guide includes "Create SonarQube Admin Token" and "Create SonarQube Token For New Project". "the token" is not clear to tell readers which one you really mean. Please adjust this part.

Reply:
They have different uses.